lib/resty/wasmx/shm.lua

```diff
@@ -355,12 +366,37 @@ local function metrics_define(zone, name, metric_type)
         error(err, 2)
     end
 
+    local cbins = nil
+    local n_bins = 0
+
+    if opts ~= nil then
+        if type(opts) ~= "table" then
+            error("opts must be a table", 2)
+        end
+
+        if opts.bins ~= nil then
```

Comment:
Should we skip this branch when the metric type is not histogram, but `opts.bins` is given? We can probably silently ignore it (as this already does, but it still creates the bins array buffer).

Reply:
Well observed. I'll make the branch to be skipped when the metric type isn't histogram.